modules/social_features/social_mentions/src/Plugin/ActivityContext/MentionActivityContext.php

```diff
@@ -45,10 +47,21 @@ public function getRecipients(array $data, $last_uid, $limit) {
       if (!empty($mentions)) {
         foreach ($mentions as $mention) {
           if (isset($mention->uid)) {
-            $recipients[] = [
-              'target_type' => 'user',
-              'target_id' => $mention->getMentionedUserId(),
-            ];
+            $uid = $mention->getMentionedUserId();
+
+            $entity_storage = \Drupal::entityTypeManager()
+              ->getStorage($mention->getMentionedEntityTypeId());
+            $mentioned_entity = $entity_storage->load($mention->getMentionedEntityId());
+
+            $account = User::load($uid);
+            $access = $mentioned_entity->access('view', $account);
```

Comment:
@jochemvn especially check out this code! Like i mentioned in the DS this morning we should also check for view access on the entity in which the user is mentioned before adding them as a recipient.